### PR TITLE
docs: make release instruction more structured

### PR DESCRIPTION
**Pull request by kpodsiad**
I hope it'll make release instruction easier to follow :)

### CODE REVIEW

File: docs/contributors/releasing.md [hunk 1]
```diff
@@ -112,18 +119,24 @@ title: Making a release
...
 
-- Update version numbers in:
-  - `build.sbt` - update `localSnapshotVersion`
-  - `.github/ISSUE_TEMPLATE/bug_report.yml` - update `Version of Metals`
-  - `.github/workflows/mtags-auto-release.yml` - update `metals_version` and `metals_ref`
-
 - Announce the new release with the link to the release notes:
   - on [Discord](https://discord.com/invite/RFpSVth)
 
+## tl;dr release steps
```
Comment (tgodzik) [inline docs/contributors/releasing.md:125]: Do we need tldr? I think it's important to go through all the step exactly as they are.

File: docs/contributors/releasing.md [hunk 2]
```diff
@@ -112,18 +119,24 @@ title: Making a release
   - Once the VS Code extension has been updated on the Marketplace, click
     "Publish release".
 
-- Update version numbers in:
-  - `build.sbt` - update `localSnapshotVersion`
```
Comment (tgodzik) [inline docs/contributors/releasing.md:116]: Why remove this? This should be done during the release ideally.

**Review by tgodzik — CHANGES_REQUESTED** [review]

File: docs/contributors/releasing.md (hunk 2 above)
Comment (kpodsiad) [reply docs/contributors/releasing.md:116]: It's duplicated, see lines 49-53

File: docs/contributors/releasing.md (hunk 1 above)
Comment (kpodsiad) [reply docs/contributors/releasing.md:125]: It's more like the sanity check, I can rename it if you don't like the `tl;dr`

File: docs/contributors/releasing.md (hunk 2 above)
Comment (kpodsiad) [reply docs/contributors/releasing.md:116]: However, this one is more concise

File: docs/contributors/releasing.md (hunk 1 above)
Comment (tgodzik) [reply docs/contributors/releasing.md]: Looks better, thanks!

**Review by tgodzik — APPROVED** [review]
LGTM

**Pull request merged**